src/librustc/middle/region.rs

```diff
@@ -1221,7 +1223,13 @@ fn resolve_local<'tcx>(
                 is_binding_pat(&subpat)
             }
 
-            _ => false,
+            PatKind::Ref(_, _) |
```

Comment:
So it never applies to `let & mut? ref mut? x` atm -- why is that?

Reply:
Because it generally wouldn't be helpful. Extending the storage lifetime of a reference doesn't extend the lifetime of what it points to.